### PR TITLE
Add logging training.

### PR DESCRIPTION
**Pull request by r0mant**
Ported from https://docs.google.com/document/d/1ZWfyQxBwtiX2KP87CF5hn-JeAD1tQglJn6RTeaNAonc/edit?usp=sharing.

### CODE REVIEW

File: logging.md [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+
+* `/var/lib/gravity/site/**/*.log`
+
+This directory contains Gravity-specific operation logs:
+
+```bash
+$ ls -l /var/lib/gravity/site/*/*.log
+```
+
+The forwarder Docker image and its configuration can be found here.
```
Comment (a-palchikov) [inline logging.md]: Which location does this refer to?

File: logging.md [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+
+## Collector
+
+Log collector is an rsyslogd server that’s running as a part of Deployment:
```
Comment (a-palchikov) [inline logging.md]: of **a** Deployment

File: logging.md [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+log-collector   ClusterIP   10.100.81.83   <none>        514/UDP,514/TCP,8083/TCP   6h54m
+```
+
+The collector itself writes all logs into `/var/log/messages` which is mounted into its pod so we can inspect all the logs on the node where the collector is running, inside the planet container:
+
+```bash
+planet$ less /var/log/messages
+```
+
+The collector Docker image and its configuration can be found here.
```
Comment (a-palchikov) [inline logging.md]: Is this a copy-paste from somewhere?

File: logging.md [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+
+The collector exposes rsyslog server via a Kubernetes Service where forwarders send entries from the files they monitor over tcp protocol:
+
+```bash
+$ kubectl -nkube-system get services/log-collector
+NAME            TYPE        CLUSTER-IP     EXTERNAL-IP   PORT(S)                    AGE
+log-collector   ClusterIP   10.100.81.83   <none>        514/UDP,514/TCP,8083/TCP   6h54m
+```
+
+The collector itself writes all logs into `/var/log/messages` which is mounted into its pod so we can inspect all the logs on the node where the collector is running, inside the planet container:
```
Comment (a-palchikov) [inline logging.md]: Worth noting a downside that if the collector `jumps`, the logs become scattered over multiple nodes.

File: logging.md [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+```
+
+The collector Docker image and its configuration can be found here.
+
+### wstail
+
+In addition to running rsyslog daemon, the log collector container also runs a program called “wstail”. It adds the following functionality:
+
+* Serves an HTTP API that allows to query all collected logs. The API is exposed via the same Kubernetes service and is used by Gravity Control Panel to provide search functionality.
+* Is responsible for creating/deleting log forwarder configurations when they are created by users.
```
Comment (a-palchikov) [inline logging.md]: when they are created/**deleted** by users.

File: logging.md [hunk 6]
```diff
@@ -0,0 +1,196 @@
...
+
+```bash
+$ kubectl -nkube-system exec log-collector-697d94486-2fgxp -- ls -l /etc/rsyslog.d
+total 4
+-rw-r--r-- 1 root root 23 Jan 15 19:22 forwarder1
+$ kubectl -nkube-system exec log-collector-697d94486-2fgxp -- cat /etc/rsyslog.d/forwarder1
+*.* @192.168.99.102:514
+```
+
+From here on out, the rsyslog server running inside the log collector pod will be forwarding all logs it receives to the configured destination using the rsyslog protocol.
```
Comment (a-palchikov) [inline logging.md]: the configured destination**s**

Comment (a-palchikov) [inline logging.md]: nit: from here on ~~out~~ (or is it still part of this?)

File: logging.md (hunk 3 above)
Comment (r0mant) [reply logging.md]: Forgot to copy links over when porting from the Google doc.

**Pull request merged**